packages/expo-updates/android/src/androidTest/java/expo/modules/updates/logging/UpdatesLoggingTest.kt

```diff
@@ -1,49 +1,81 @@
-    Assert.assertNull(entryCopy.updateId)
-    Assert.assertNull(entryCopy.assetId)
-    Assert.assertNull(entryCopy.stacktrace)
+    Assert.assertEquals(entry.message, entryCopy?.message)
+    Assert.assertEquals(entry.timestamp, entryCopy?.timestamp)
+    Assert.assertEquals(entry.code, entryCopy?.code)
+    Assert.assertEquals(entry.level, entryCopy?.level)
+    Assert.assertNull(entryCopy?.updateId)
+    Assert.assertNull(entryCopy?.assetId)
+    Assert.assertNull(entryCopy?.stacktrace)
 
     val entry2 = UpdatesLogEntry(12345678, "Test message", "UpdateFailedToLoad", "fatal", "myUpdateId", "myAssetId", listOf("stack frame 1", "stack frame 2"))
     val json2 = entry2.asString()
     val entryCopy2 = UpdatesLogEntry.create(json2)
-    Assert.assertEquals(entry2.message, entryCopy2.message)
-    Assert.assertEquals(entry2.timestamp, entryCopy2.timestamp)
-    Assert.assertEquals(entry2.code, entryCopy2.code)
-    Assert.assertEquals(entry2.level, entryCopy2.level)
-    Assert.assertEquals(entry2.updateId, entryCopy2.updateId)
-    Assert.assertEquals(entry2.assetId, entryCopy2.assetId)
-    Assert.assertNotNull(entryCopy2.stacktrace)
-    Assert.assertEquals(entry2.stacktrace?.size, entryCopy2.stacktrace?.size)
+    Assert.assertEquals(entry2.message, entryCopy2?.message)
+    Assert.assertEquals(entry2.timestamp, entryCopy2?.timestamp)
+    Assert.assertEquals(entry2.code, entryCopy2?.code)
+    Assert.assertEquals(entry2.level, entryCopy2?.level)
+    Assert.assertEquals(entry2.updateId, entryCopy2?.updateId)
+    Assert.assertEquals(entry2.assetId, entryCopy2?.assetId)
+    Assert.assertNotNull(entryCopy2?.stacktrace)
+    Assert.assertEquals(entry2.stacktrace?.size, entryCopy2?.stacktrace?.size)
+
+    // Test that invalid JSON converts to null
+    val testInvalidJSON = "{\"timestamp\":1600000,\"message\":\"Test message\",\"code\":\"JSRuntimeError\",\"level\":\"wa"
+    Assert.assertNull(UpdatesLogEntry.create(testInvalidJSON))
+
+    // Test that valid JSON missing a required field converts to null
+    val testMissingRequiredFieldJSON = "{\"message\":\"Test message\",\"code\":\"JSRuntimeError\",\"level\":\"warn\"}"
+    Assert.assertNull(UpdatesLogEntry.create(testMissingRequiredFieldJSON))
   }
 
   @Test
   fun testOneLogAppears() {
-    val logger = UpdatesLogger()
-    logger.warn("Test message", UpdatesErrorCode.JSRuntimeError)
+    val instrumentationContext = InstrumentationRegistry.getInstrumentation().context
+    val logger = UpdatesLogger(instrumentationContext)
     val now = Date()
-    val expectedLogEntry = UpdatesLogEntry(now.time, "Test message", UpdatesErrorCode.JSRuntimeError.code, UpdatesLogType.Warn.type, null, null, null)
+    val expectedLogEntry = UpdatesLogEntry(now.time, "Test message", UpdatesErrorCode.JSRuntimeError.code, LogType.Warn.type, null, null, null)
+    logger.warn("Test message", UpdatesErrorCode.JSRuntimeError)
+    asyncTestUtil.waitForTimeout(500)
```

Comment:
we'll want to create a unique asyncTestUtil per test to fully address my previous comment

Reply:
We are doing that in setup()

Reply:
I've fixed this for both test classes now.